sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpSender.cs

```diff
@@ -491,16 +497,17 @@ public override async Task CloseAsync(CancellationToken cancellationToken)
             long[] sequenceNumbers,
             CancellationToken cancellationToken = default)
         {
-            Task cancelMessageTask = _retryPolicy.RunOperation(async (timeout) =>
-            {
-                await CancelScheduledMessageInternalAsync(
-                    sequenceNumbers,
-                    timeout,
-                    cancellationToken).ConfigureAwait(false);
-            },
-            _connectionScope,
-            cancellationToken);
-            await cancelMessageTask.ConfigureAwait(false);
+            await _retryPolicy.RunOperation(static async (value, timeout, token) =>
+                {
+                    var (sender, sqn) = value;
```

Comment:
nit  The rename of  `sequenceNumbers` to `sqn` doesn't convey "this is a set of things" to me.    Can we restore the original name?

Reply:
One of the drawbacks this approach has is that we have to redeclare names as variables in the inner scope. So I chose names that more or less phonetically resemble the outers but still they have to be different. So this seems to be one of the choices between readibility and perf. I'm open for suggestions

Reply:
maybe just suffix each variable with "inner"?

Reply:
What about prefixing with underscores?

Reply:
I don't think so as those are used for private instance members.

Reply:
Double underscores :trollface: ? Just kidding. I completely forgot that with static lambdas I can actually use the same names as the outer methods and we won't clash. I changed to that for now